src/core/SuccessResponse.ts

```diff
@@ -11,4 +11,21 @@ export type Meta<T> = {
     requestHeaders: { contentType: string | null }
 }
 
-export type SuccessResponse<T> = T & { _meta: Meta<T> }
+type AsObject<T> = T extends object ? (T extends any[] ? { asObject: T } : T) : { asObject: T }
```

Comment:
I'm not sure if `asObject` sits right with me, reasons:

* It's named after an implementation details (we're converting non-pure-object to an object) instead of the semantics (the property holds X)
* Counterintuitively the property named `asObject` does not hold a regular object, just arrays, potentially strings etc.

Alternatives, although not great:

* `data`
* Name the property differently depending on the type:
  * array -> `items`, `array`, `rows`, `elements`
  * non-array -> `data`

Reply:
Concerning the name of the property, not the `AsObject` type.

Reply:
`notAnObject`